lib/elsa/producer.ex

```diff
@@ -68,9 +68,14 @@ defmodule Elsa.Producer do
     end
   end
 
-  defp transform_message(%{key: _key, value: _value} = msg), do: msg
-  defp transform_message({key, value}), do: %{key: key, value: value}
-  defp transform_message(message), do: %{key: "", value: message}
+  defp transform_message(%{key: _key, value: _value} = msg) do
+    msg
+    |> Map.update!(:key, fn key -> IO.iodata_to_binary(key) end)
+    |> Map.update!(:value, fn value -> IO.iodata_to_binary(value) end)
```

Comment:
```suggestion
    |> Map.update!(:key, &IO.iodata_to_binary/1)
    |> Map.update!(:value, &IO.iodata_to_binary/1)
```

Nitpick-y but I think it's an improvement.